compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -80,6 +94,33 @@
                                     PlatformLibraryScope.DEFAULT,
                                     getPackageName(rootPackageContext)));
 
+        // TODO: Move to a compiler extension once Compiler revamp is complete
+        // 4) Add the Observability Symbols Jar
+        if (rootPackageContext.compilationOptions().observabilityIncluded()) {
+            Manifest manifest = new Manifest();
+            manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
+
+            try {
+                String packageName = getPackageName(rootPackageContext);
+                Path jarPath = Path.of(System.getProperty("java.io.tmpdir"),
+                        packageName.replaceAll(File.separatorChar == '\\' ? "\\\\" : File.separator, "-")
+                                + "-" + UUID.randomUUID() + ".jar");
+                JarOutputStream jarOutputStream = new JarOutputStream(new BufferedOutputStream(
+                        new FileOutputStream(jarPath.toFile())), manifest);
+                jarOutputStream.close();
```

Comment:
What is the purpose of creating a jarOutputStream and closing it without writing any entries?

Reply:
This is to generate a proper empty (still it will contain the manifest) Jar file which can be used by the ObservabilitySymbolCollector to store the syntax tree.

Reply:
Added a comment in fe87a0066595c9fa06d4693306192fa1e3d80386 to improve readability